src/v/security/audit/client.cc

```diff
@@ -28,6 +29,151 @@ namespace security::audit {
 
+class rpc_client_impl final : public audit_client {
+public:
+    rpc_client_impl(
+      audit_sink* sink,
+      cluster::controller* controller,
+      kafka::data::rpc::client* rpc_client)
+      : audit_client(sink, controller)
+      , _rpc_client(rpc_client) {}
+    rpc_client_impl(const rpc_client_impl&) = delete;
+    rpc_client_impl& operator=(const rpc_client_impl&) = delete;
+    rpc_client_impl(rpc_client_impl&&) = delete;
+    rpc_client_impl& operator=(rpc_client_impl&&) = delete;
+    ~rpc_client_impl() final = default;
+
+    ss::future<> do_produce(
+      model::record_batch batch,
+      model::partition_id pid,
+      audit_probe& probe,
+      ss::timer<>::time_point timeout) final {
+        // TODO(oren): map cluster errc to kafka. we should be able to do that
+        // actually with better fidelity. but really this is mostly for
+        // detecting auth errors I think, which is totally unnecessary here.
+        constexpr auto map_ec = [](cluster::errc ec) -> kafka::error_code {
+            if (ec == cluster::errc::success) {
+                return kafka::error_code::none;
+            }
+            return kafka::error_code::unknown_server_error;
+        };
+
+        std::optional<kafka::error_code> ec;
+        while (!as().abort_requested() && ss::timer<>::clock::now() < timeout) {
+            auto r = co_await _rpc_client->produce(
+              model::topic_partition{model::kafka_audit_logging_topic, pid},
+              batch.copy());
+            ec.emplace(map_ec(r));
+            if (ec.value() == kafka::error_code::none) {
+                break;
+            }
+        }
+
+        // report unknown server error if we aborted before making any attempt
+        if (auto errc = ec.value_or(kafka::error_code::unknown_server_error);
+            errc != kafka::error_code::none) {
+            vlog(
+              adtlog.warn,
+              "{} audit records dropped, shutting down. Last error: {}",
+              batch.record_count(),
+              errc);
+            probe.audit_error();
+        } else {
+            probe.audit_event();
+        }
+    }
+    ss::future<> client_shutdown() final {
+        /// Nothing to do for the rpc client
+        return ss::now();
+    }
+    ss::future<> create_internal_topic() {
+        // constexpr std::string_view seven_days = "604800000";
+        using namespace std::chrono_literals;
+
+        int16_t replication_factor
+          = config::shard_local_cfg().audit_log_replication_factor().value_or(
+            controller()->internal_topic_replication());
+        vlog(
+          adtlog.debug,
+          "TODO(plumbing): Attempting to create internal topic "
+          "(replication={})",
+          replication_factor);
+
+        cluster::topic_properties audit_topic_props;
+        audit_topic_props.retention_bytes = tristate<size_t>{};
+        // TODO(oren): respect the replication config, otherwise it uses the
+        // default
```

Comment:
pretty sure @michael-redpanda added support for this is a recent PR (or a pending one?)

Reply:
it's here: https://github.com/redpanda-data/redpanda/pull/27245/commits/51dacee13df34d4e7499caff3b91d55a86c7614d